projects/SmallTheftAuto/Assets/Scripts/CarMovement.cs

```diff
@@ -4,12 +4,20 @@
 
 public class CarMovement : MonoBehaviour {
 
+    /*
+     * Instead of public variables, I rather keep it private but serializable.
```

Comment:
The fields were made public by design if we wanted another script to set the values instead of manually inputing in the Unity editor (e.g. future updates with different car types)

Reply:
In this case the best practice is to use

```cs
public int Health {get; set;}
```

It is called encapsulation, I couldn't find the slides, but Marc did show it to us in the ObjectOrientedProgram week.

Were a link explaining why we should design this way :) :
[https://www.c-sharpcorner.com/article/encapsulation-in-C-Sharp/](https://www.c-sharpcorner.com/article/encapsulation-in-C-Sharp/)

Reply:
Here is the exercise we did using encapsulation:
[https://github.com/marczaku/csharp-oop/blob/main/assignments/003.5.1-console-classes-1.md#11---properties](https://github.com/marczaku/csharp-oop/blob/main/assignments/003.5.1-console-classes-1.md#11---properties)

Reply:
The tag [SerializeField] ensures that private variables can be changed in the Editor